### ISSUE TITLE
Possible missinformation about Context and MobXProviderContext

### ISSUE THREAD

**Issue by erhathaway** [mobxjs/mobx-react#726]
The documentation says
> Note: usually there is no need anymore to use Provider / inject in new code bases; most of its features are now covered by React.createContext.

However, it is not as straight forward to use `observer` with `Context` as it was to use `inject` for reasons documented https://github.com/mobxjs/mobx-react/issues/471

The docs say to use `MobXProviderContext`:
>It is possible to read the stores provided by Provider using React.useContext, by using the MobXProviderContext context that can be imported from mobx-react.

Also, discussion in the above issue has some good suggestions for what this context proxy should look like, however it doesn't seem to work. In the code base `MobXProviderContext` simply aliases `export const MobXProviderContext = createContext({})`

Furthermore, in #689 there was talk of
>  The MobXProviderContext is only useful for code using inject and migrating to hooks. 

There seems to be a missmatch in what the docs suggest and the implementation of the code.  I get how `MobXProviderContext` is used with `Provider`, but I don't understand why this should replace `React.createContext`, as the docs suggest.

I feel like this suggestion should be removed from the docs and instead info around why using the `Observer` component inside a `Context.consumer` should be added. Additionally a full implementation of `MobXProviderContext`, as suggested in #471, would probably be helpful to others.

I'd be happy to makes some PRs to get this rolling. 

Mainly, I'm curious if I'm missing something?

**Comment by FredyC** [comment]
Well, #471 is somewhat old in terms that Hooks came out in 2019 and changed a view on a lot of things. So I wouldn't consider comments there too relevant today. But of course, if you are stuck with class components for time being, it still might hold some relevance. However, instead of `Consumer` component, the [static `contextType`](https://reactjs.org/docs/context.html#classcontexttype) can be a better option. 

It's true that docs are mentioning `MobXProviderContext` in relation to Hooks only which might feel slightly misleading.

I am slightly confused about why you don't want to keep using `inject`. For class components it probably still a better solution. And since V6 it uses Context underneath so there is no problem for a future React versions.

I wonder why would anyone go through the trouble of refactoring class component to use Context. Isn't it better to refactor to function component right away?

> but I don't understand why this should replace `React.createContext`, as the docs suggest.

Because the concept of `createContext` is so simple that there is no need to have it in the library. The future proof `mobx-react-lite` has already dropped it. I wrote a bunch of information on [how to utilize Context in function components](https://mobx-react.js.org/recipes-context) and [why is inject obsolete](https://mobx-react.js.org/recipes-inject).

**Comment by erhathaway** [comment]
@FredyC thanks for the quick response and the posts you linked to!

My concern mainly boils down to the docs saying that `context` should be used over `inject`. If someone is using a class component (for whatever reason) and tries to use context, it is somewhat unclear that you can't just use `observe` with the  `Context.consumer`  - you need to use `Observe` and that isn't mentioned in the scope of the same section.

**Comment by FredyC** [comment]
> it is somewhat unclear that you can't just use `observe` with the `Context.consumer` - you need to use `Observe` and that isn't mentioned in the scope of the same section.

Yes, you are right, that is surely confusing. I will accept a PR if you have an idea how to improve that. Possibly recommend the `contextType` I've linked above.

**Comment by urugator** [comment]
> My concern mainly boils down to the docs saying that context should be used over inject.

The `inject` originally used experimental (legacy) context. Now since the context API is stable, it's preferred to use it directly (with all it's inconveniences).

> it is somewhat unclear that you can't just use observe with the Context.consumer

Imo the docs should suggest to either use `contextType` or to abstract consuming logic into a HOC like `withMyContext(observer(Component))`
The direct use of `<Context.Consumer>` should be avoided.
